c2rust-analyze/src/context.rs

```diff
@@ -41,6 +41,70 @@ bitflags! {
         const OFFSET_SUB = 0x0020;
         /// This pointer can be freed.
         const FREE = 0x0040;
+
+        /// This pointer is non-null.
+        ///
+        /// [`NON_NULL`] is set (or not) when the pointer is created,
+        /// and it flows forward along dataflow edges.
+        ///
+        /// The permission is [`NON_NULL`] rather than `NULL` because we allow dropping permissions in any assignment.
+        /// This means removing a permission from a pointer's [`PermissionSet`]
+        /// must allow the pointer to take on more values, not restrict it to fewer values.
```

Comment:
This bit from "we allow dropping permissions" onward applies to all permissions - could you move it to `PermissionSet`'s doc comment instead of `NON_NULL`'s?

Reply:
Sure, just this paragraph, right?

Reply:
Right, just this paragraph.

Reply:
Also fixed in 1dde612baa11ac02692f9eafd8f5fe9f2b6cd7a2.